### PR TITLE
Modernizing for Azure Functions OpenAI extension bindings, Flex Consumption, and Managed Identity

### PR DESCRIPTION
**Pull request by paulyuk**
## Purpose
- Updated the code to use the Azure Functions OpenAI extension bindings
- Bicep now uses Azure Functions Flex Consumption and Managed Identity (no more keys/secrets)
- Added Azure Functions DevContainer/Codespaces support

## Does this introduce a breaking change?

```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[x] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
Follow the readme

### CODE REVIEW

**Comment by paulyuk** [comment]
@dm-chelupati could you please review?

**Comment by paulyuk** [comment]
@vrdmr could you please review?

**Review by dm-chelupati — APPROVED** [review]
LGTM

**Review by dm-chelupati — APPROVED** [review]
LGTM

File: infra/app/api.bicep [hunk 1]
```diff
@@ -1,38 +1,50 @@
...
+      })
     applicationInsightsName: applicationInsightsName
     appServicePlanId: appServicePlanId
-    keyVaultName: keyVaultName
-    //py
-    numberOfWorkers: 1
-    minimumElasticInstanceCount: 0
-    //--py
-    runtimeName: 'python'
-    runtimeVersion: '3.9'
```
Comment (dm-chelupati) [inline infra/app/api.bicep:30]: why not python 3.11?

Comment (paulyuk) [reply infra/app/api.bicep:30]: fixing now.  thank you for review!

Comment (paulyuk) [reply infra/app/api.bicep:30]: oh. that's deleted lines.  I had fixed it in `main.bicep`.  This sample migrated from Consumption (v1) hosting plan to Flex Consumption.  Language stack version is handled differently now, as a part of the resource creation and not the config.

Snip from PR's `main.bicep`
```json
module api './app/api.bicep' = {
  name: 'api'
  scope: rg
  params: {
    name: functionAppName
    location: location
    tags: tags
    applicationInsightsName: monitoring.outputs.applicationInsightsName
    appServicePlanId: appServicePlan.outputs.id
    runtimeName: 'python'
    runtimeVersion: '3.11'
    ```

**Pull request merged**